### ISSUE TITLE
Dune installs empty META file for package with no library

### ISSUE THREAD

**Issue by dra27** [ocaml/dune#2353]
If a package defines executables only, Dune generates a `META` file containing a blank line and installs it. cppo, since its conversion to Jbuilder in v1.6.0 has had this `cppo.install` generated:

```
lib: [
  "_build/install/default/lib/cppo/META"
  "_build/install/default/lib/cppo/dune-package"
  "_build/install/default/lib/cppo/opam"
]
bin: [
  "_build/install/default/bin/cppo"
]
doc: [
  "_build/install/default/doc/cppo/LICENSE.md"
  "_build/install/default/doc/cppo/README.md"
]
```

but because `META` is empty, you get this from `ocamlfind list`:
```
cppo                (version: n/a)
cppo_ocamlbuild     (version: 1.6.6)
```

Dune should either not generate a `META` file at all or should put at least a `version` field in it. If `META` is written, then I think a `synopsis` field might also be wanted for `executable` which would go in `description` in the `META` file.

I have no opinion on whether there should be a `META` file at all (prior to 1.6.0, cppo didn't have one, unsurprisingly).

**Comment by diml** [comment]
We should indeed generate no `META` and no `dune-package` file

**Comment by dra27** [comment]
I agree - although note from the linked PR that that change is detectable, so it *might* want to be tied to a specific dune language version?

**Comment by diml** [comment]
Indeed. Tying it to a specific dune language version seems good
